pkgs/applications/misc/syncthingtray/default.nix

```diff
@@ -0,0 +1,50 @@
+{ mkDerivation, stdenv, lib, fetchFromGitHub
+, qtbase, extra-cmake-modules, cpp-utilities, qtutilities
+, cmake, kio, plasma-framework, qttools
+, webviewProvider ? null
+, jsProvider ? null
+, enableKioPluginSupport ? false
+, enablePlasmoidSupport  ? false
+, systemdSupport ? true }:
+
+mkDerivation rec {
+  version = "0.9.1";
+  pname = "syncthingtray";
+
+  src = fetchFromGitHub {
+    owner  = "Martchus";
+    repo   = pname;
+    rev    = "v${version}";
+    sha256 = "0ijwpwlwwbfh9fdfbwz6dgi6hpmaav2jm56mzxm6as50iwnb59fx";
+  };
+
+  buildInputs = [ qtbase cpp-utilities qtutilities webviewProvider jsProvider ] 
+    ++ lib.optional enableKioPluginSupport kio
+    ++ lib.optional enablePlasmoidSupport [ extra-cmake-modules plasma-framework ]
+  ;
+
+  nativeBuildInputs = [ cmake qttools ];
+
+  cmakeFlags = lib.optionals (enablePlasmoidSupport == false) ["-DNO_PLASMOID=ON"]
+    ++ lib.optionals (enableKioPluginSupport == false) ["-DNO_FILE_ITEM_ACTION_PLUGIN=ON"]
+    ++ lib.optionals systemdSupport ["-DSYSTEMD_SUPPORT=ON"]
+  ;
+  # Without this hook, `make install` fails since it tries to write to ${qtbase.out}/lib/qt-<version>/...
+  # This is kind of ugly, but it works and it'll probably be quicker than to
+  # bother the author of this program as he's using arch Linux and he'll probably be surprised that we
+  # modify the prefix of our packages and not using `make DESTDIR= install`.
+  preInstall = ''
+    echo grepping for ${qtbase.out} as \$qtbase in install files inside the build directory
+    grep -l -R ${qtbase.out}/lib/qt- | while read f; do
+      substituteInPlace $f --replace ${qtbase.out} $out
+    done || :
```

Comment:
Why the "or true"?

Reply:
You are right, I've had it stayed there from my tests with `nix-shell`.